Fix nullable dereference warning when reading identity cookie name

## Summary
`dotnet build` reports CS8602 in `Program.cs` when reading `Cookie.Name` from the identity application cookie options.

## Problem
The build succeeds but emits a nullable warning:
- `Program.cs(178,59): warning CS8602: Dereference of a possibly null reference.`

## Proposed fix
Use an explicit null-safe read of the cookie name so startup gate cookie filtering remains unchanged and the warning is removed.

## Reproduction
1. `dotnet build ./src/WebApp/PingMonitor.Web/PingMonitor.Web.csproj --configuration Release`
2. Observe CS8602 warning in build output.

--- a/src/WebApp/PingMonitor.Web/Program.cs
+++ b/src/WebApp/PingMonitor.Web/Program.cs
@@ -151,7 +151,7 @@ builder.Services.AddHostedService(sp => sp.GetRequiredService<ConfigurationAutoB
 var app = builder.Build();
 
 var identityCookieOptionsMonitor = app.Services.GetRequiredService<IOptionsMonitor<CookieAuthenticationOptions>>();
-var identityApplicationCookieName = identityCookieOptionsMonitor.Get(IdentityConstants.ApplicationScheme).Cookie.Name;
+var identityApplicationCookieName = identityCookieOptionsMonitor.Get(IdentityConstants.ApplicationScheme).Cookie?.Name;
 
 app.UseForwardedHeaders();
 app.UseHttpsRedirection();
@@ -175,9 +175,13 @@ app.Use(async (context, next) =>
             context.Request.Headers.TryGetValue("Cookie", out var cookieHeaderValues))
         {
             var filteredCookieHeaders = cookieHeaderValues
-                .Select(cookieHeader => string.Join("; ", cookieHeader
-                    .Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
-                    .Where(cookiePart => !cookiePart.StartsWith($"{identityApplicationCookieName}=", StringComparison.Ordinal))))
+                .Select(cookieHeader =>
+                {
+                    var nonNullCookieHeader = cookieHeader ?? string.Empty;
+                    return string.Join("; ", nonNullCookieHeader
+                        .Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
+                        .Where(cookiePart => !cookiePart.StartsWith($"{identityApplicationCookieName}=", StringComparison.Ordinal)));
+                })
                 .Where(cookieHeader => !string.IsNullOrWhiteSpace(cookieHeader))
                 .ToArray();
 

